sdk/src/main/scala/io/horizen/api/http/route/SidechainNodeApiRoute.scala

```diff
@@ -123,6 +171,57 @@ case class SidechainNodeApiRoute(peerManager: ActorRef,
     }
   }
 
+  def addToBlacklist: Route = (post & path("blacklist") & withBasicAuth) {
+    _ => {
+      try {
+        entity(as[RegAddToBlacklist]) { bodyRequest =>
+          val peerAddress = bodyRequest.address
+          val banDuration = bodyRequest.duration
+
+          if (banDuration <= 0) {
+            SidechainApiError(s"duration must be greater than 0; $banDuration not allowed")
+          } else {
+            addressAndPortRegexp.findFirstMatchIn(peerAddress) match {
+              case None => SidechainApiError(s"address $peerAddress is not well formatted")
+
+              case Some(addressAndPort) =>
+                val host = InetAddress.getByName(addressAndPort.group(1))
+                val port = addressAndPort.group(2).toInt
+                val peerAddress = new InetSocketAddress(host, port)
+                peerManager ! AddToBlacklist(peerAddress, Some(CustomPenaltyDuration(banDuration)))
+                networkController ! DisconnectFromAddress(peerAddress)
+                ApiResponse.OK
+            }
+          }
+        }
+      } catch {
+        case e: Throwable => SidechainApiError(e)
+      }
+    }
+  }
+
+  def removeFromBlacklist: Route = (path("blacklist") & delete & withBasicAuth) {
```

Comment:
Done
Route changed to `removeFromBlacklist` because of conflict with `addToBlacklist` method. The latter's route was also changed to `addToBlacklist` for consistence in paths.